src/clojure/nrepl/middleware/dynamic_loader.clj

```diff
@@ -0,0 +1,108 @@
+(ns nrepl.middleware.dynamic-loader
+  "Support the ability to interactively update the middleware of the *running*
+  nREPL server. This can be used by tools to configure an existing instance of
+  an environment after connection.
+
+  When combined with the sideloader, this could be used to inject middlewares
+  that are unknown to the server prior to connection."
+  {:author "Shen Tian", :added "0.8"}
+  (:require [clojure.string :as str]
+            [nrepl.middleware :as middleware :refer [set-descriptor!]]
+            [nrepl.middleware.session :as middleware.session]
+            [nrepl.misc :refer [response-for]]
+            [nrepl.transport :as t]))
+
+(defn require-resolve
+  [sym-str]
+  (let [sym (symbol sym-str)
+        ns (namespace sym)]
+    (when-not (find-ns sym)
+      (require (symbol ns))
+      (resolve sym))))
+
+(def ^:dynamic *state* nil)
+
+(defn wrap-ping
+  [h]
+  (fn [{:keys [op transport] :as msg}]
+    (case op
+      "ping"
+      (t/send transport (response-for msg {:pong "pong"
+                                           :status :done}))
+      (h msg))))
+
+(set-descriptor! #'wrap-ping
```

Comment:
I guess that's some test middleware, right?

Reply:
Yes. Will remove before asking for a full review